### PR TITLE
Add the sdbus-cpp package to Mariner

### PR DESCRIPTION
**Pull request by snambakam**
Source: https://github.com/Kistler-Group/sdbus-cpp

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR adds the sdbus-cpp package to Mariner.

It is required by some teams to communicate with dbus.

###### Change Log  

N/A

###### Does this affect the toolchain?  
NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

local build

### CODE REVIEW

File: SPECS/sdbus-cpp/sdbus-cpp.spec [hunk 1]
```diff
@@ -0,0 +1,71 @@
+Summary:        sdbus-cpp
+Name:           sdbus-cpp
+Version:        1.1.0
+Release:        0%{?dist}
+License:        LGPLv2+
+URL:            https://github.com/Kistler-Group/sdbus-cpp
+Source0:        %{name}-%{version}.tar.bz2
```
Comment (mfrw) [inline SPECS/sdbus-cpp/sdbus-cpp.spec:7]: I think we might have missed to add an entry to `cgmanifest.json` :)

File: SPECS/sdbus-cpp/sdbus-cpp.spec [hunk 2]
```diff
@@ -0,0 +1,71 @@
+Summary:        sdbus-cpp
+Name:           sdbus-cpp
+Version:        1.1.0
+Release:        0%{?dist}
+License:        LGPLv2+
```
Comment (mfrw) [inline SPECS/sdbus-cpp/sdbus-cpp.spec:5]: We should add an entry to the `SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`.
A representative PR would be: 
- #2964

File: SPECS/sdbus-cpp/sdbus-cpp.signatures.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+ "Signatures": {
+  "sdbus-cpp-1.1.0.tar.bz2": "62eef2197d5079206e103573614feee72c6831fe599a1f5880413a64e8b8fca7"
```
Comment (mfrw) [inline SPECS/sdbus-cpp/sdbus-cpp.signatures.json:3]: We might have missed to upload the `tar.gz` to the blob store.

File: SPECS/sdbus-cpp/sdbus-cpp.spec [hunk 4]
```diff
@@ -0,0 +1,71 @@
+Summary:        sdbus-cpp
+Name:           sdbus-cpp
+Version:        1.1.0
+Release:        0%{?dist}
```
Comment (mfrw) [inline SPECS/sdbus-cpp/sdbus-cpp.spec:4]:
```suggestion
Release:        1%{?dist}
```

File: SPECS/sdbus-cpp/sdbus-cpp.spec [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+%files devel
+%defattr(-,root,root,-)
+%{_includedir}/sdbus-c++/*.h
+%{_includedir}/sdbus-c++/*.inl
+%{_libdir}/cmake/sdbus-c++/*.cmake
+%{_libdir}/pkgconfig/*.pc
+
+%changelog
+*   Tue Feb 8 2022 Sriram Nambakam <snambakam@microsoft.com> 1.1.0-1
```
Comment (mfrw) [inline SPECS/sdbus-cpp/sdbus-cpp.spec:70]:
```suggestion
*   Tue Feb 8 2022 Sriram Nambakam <snambakam@microsoft.com> - 1.1.0-1
```

Comment (snambakam) [reply SPECS/sdbus-cpp/sdbus-cpp.spec]: Fixed

File: SPECS/sdbus-cpp/sdbus-cpp.spec (hunk 4 above)
Comment (snambakam) [reply SPECS/sdbus-cpp/sdbus-cpp.spec]: Fixed

File: SPECS/sdbus-cpp/sdbus-cpp.spec (hunk 2 above)
Comment (snambakam) [reply SPECS/sdbus-cpp/sdbus-cpp.spec:5]: Fixed

File: SPECS/sdbus-cpp/sdbus-cpp.spec [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+%files devel
+%defattr(-,root,root,-)
+%{_includedir}/sdbus-c++/*.h
+%{_includedir}/sdbus-c++/*.inl
+%{_libdir}/cmake/sdbus-c++/*.cmake
+%{_libdir}/pkgconfig/*.pc
+
+%changelog
+*   Tue Feb 8 2022 Sriram Nambakam <snambakam@microsoft.com> - 1.1.0-1
+-   Initial build. First version
```
Comment (neha170) [inline SPECS/sdbus-cpp/sdbus-cpp.spec:71]: Please put the first changelog as '- Initial CBL-Mariner import from XX (license: XX)', or 'Original version for CBL-Mariner' for the linter, thanks

File: SPECS/sdbus-cpp/sdbus-cpp.signatures.json (hunk 3 above)
Comment (nicogbg) [inline SPECS/sdbus-cpp/sdbus-cpp.signatures.json]: you should update the signature and the extension (.tar.gz) since you're now using tar.gz file

**Review by nicogbg — APPROVED** [review]

File: SPECS/sdbus-cpp/sdbus-cpp.signatures.json (hunk 3 above)
Comment (snambakam) [reply SPECS/sdbus-cpp/sdbus-cpp.signatures.json]: Fixed

File: SPECS/sdbus-cpp/sdbus-cpp.spec (hunk 6 above)
Comment (snambakam) [reply SPECS/sdbus-cpp/sdbus-cpp.spec]: Did this and saw same errors.

**Pull request merged**